dask/dataframe/core.py

```diff
@@ -4942,3 +4944,58 @@ def meta_warning(df):
                 "  Before: .apply(func)\n"
                 "  After:  .apply(func, meta=%s)\n" % str(meta_str))
     return msg
+
+
+def mapseries(base_chunk, concat_map):
+    return base_chunk.map(concat_map.sort_index())
```

Comment:
Why the `sort_index`?

Reply:
the `concat_map` will not come in sorted, and pandas `map` operation errors if the index of the mapping series is not sorted. This ensures it's sorted so pandas map is happy.

Reply:
> pandas map operation errors if the index of the mapping series is not sorted.

Can you clarify. IIUC, that's not correct

```python
In [9]: s = pd.Series([3, 2, 1], index=[1, 0, 2])

In [10]: t = pd.Series([2, 1, 3], index=['b', 'a', 'c'])

In [11]: t.map(s)
Out[11]:
b    1.0
a    3.0
c    NaN
dtype: float64
```

Reply:
All in all: The tests here pass at the moment without `.sort_index`, and I can't seem to be able to reproduce this odd `is_unique` bug, but I suspect it might come back. 

so, for now: I removed the `.sort_index()` and added some tests to the series_map that shuffles the index of the mapper as well.